spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/AggregatePath.java

```diff
@@ -34,7 +43,7 @@
  * @author Jens Schauder
  * @author Mark Paluch
  */
-public interface AggregatePath extends Iterable<AggregatePath> {
+public interface AggregatePath extends Iterable<AggregatePath>, Comparable<AggregatePath> {
```

Comment:
What is the motivation for making this `Comparable`? What are the rules for comparison? How can one `AggregatePath` be smaller/greater than another one?

Reply:
It's for allowing ordering `AggregatePath` instances in a consistent way. The actual ordering isn't that important and the implementation is based on the comparison of the dot-notation of the path.

Added som clarifying comments.